src/Models/Audit.php

```diff
@@ -19,6 +19,6 @@ class Audit extends Model implements \OwenIt\Auditing\Contracts\Audit
     protected $casts = [
         'old_values'   => 'json',
         'new_values'   => 'json',
-        'auditable_id' => 'integer',
+        // Note: Please do not add 'auditable_id' in here, as it will break non-integer PK models
```

Comment:
Why does this break non-integer PKs? If you override the model in your code to use another type of PK, you can also override the `$casts`. In my opinion, this should not be changed as it would force others (like me) that are using the default configuration to add this cast manually in a customized model.

Reply:
@Namoshek This is a vendor model, you can't just override it. The reason this breaks non-integer PKs is because it casts UUID PKs (or any stirng PKs) as integer, thereby changing them completely. I added a test in my PR which demonstrates this.

The cast shouldn't be necessary in any case - because auditable_id could be of any type in theory (but in particular integer or string), so this package shouldn't make assumptions as to it's type - especially given Laravel/Eloquent itself supports string key types directly.

Reply:
Of course you can override it. It is the first option in the config:

https://github.com/owen-it/laravel-auditing/blob/eff1fe27f3be42eeafbd72ffb9ea55b37a3142be/config/audit.php#L5-L14

Reply:
@Namoshek So can you explain why someone would need to override this model for acquire it's compatibility with default Laravel functionality?

If you have some special case where it doesn't work for you, then perhaps you should override it yourself, but the way it is in my PR works for all keys of all major databases (Postgres, MySQL, MariaDB, etc).

Reply:
Yes, I do have a reasoning why it should be an integer cast:
- The default migration is for an integer column. If you adjust the migration, you can or should also adjust the model.
- The default Eloquent behavior is an integer primary key.
- The majority of users will use integer primary keys throughout their whole application, making it the obvious choice for the more suitable default.

In the end it is the decision of the maintainer. He did announce the change in the changelog, although I have to admit that it should have been a minor release as it may break BC as in your case.

Reply:
@Namoshek The fact that it's "default" makes no difference, as both are equally supported functionality, and it will work for both without casting.

You can not go making assumptions what "the majority" of users are using, and it is not relevant anyway - so long as it's a core feature of Laravel, this package shouldn't break that.